src/CapabilityService.WebApi/Infrastructure/Persistence/TransactionalDecoratorBase.cs

```diff
@@ -0,0 +1,39 @@
+using System;
+using System.Threading.Tasks;
+
+namespace DFDS.CapabilityService.WebApi.Infrastructure.Persistence
+{
+    public abstract class TransactionalDecoratorBase
+    {
+        private readonly CapabilityServiceDbContext _dbContext;
+
+        protected TransactionalDecoratorBase(CapabilityServiceDbContext dbContext)
+        {
+            _dbContext = dbContext;
+        }
+
+        protected async Task ExecuteInTransaction(Func<Task> action)
```

Comment:
Refactor to the below to be compliant with @ https://docs.microsoft.com/en-us/ef/core/miscellaneous/connection-resiliency#execution-strategies-and-transactions